File: tests/cpp/src/Stream.cpp
/*
 * CAMEO
 *
 * Copyright 2015 Institut Laue-Langevin
 *
 * Licensed under BSD 3-Clause and GPL-v3 as described in license files.
 * You may not use this work except in compliance with the Licences.
 *
 */

#include <iostream>
#include <string>
#include <vector>
#include <sstream>
#include <cameo/api/cameo.h>

using namespace std;
using namespace cameo;

int main(int argc, char *argv[]) {

	This::init(argc, argv);

	This::setRunning();

	int i = 0;
	while (i < 20) {
		cout << "Printing " << i << endl;
		this_thread::sleep_for(chrono::milliseconds(100));
		i++;
	}

	cout << "Finished the application" << endl;

	return 0;
}

File: tests/python/src/stream.py
import sys
import time
import cameopy

this = cameopy.This
this.init(sys.argv)

this.setRunning()

for i in range(20):
    print("Printing " + str(i))  
    time.sleep(0.1)

this.terminate()

print("Finished the application")
